packages/node/src/node.ts

```diff
@@ -110,10 +113,13 @@ export class Node extends Base {
         const key = item[0];
         const value = item[1];
 
-        const result = this.sanitizeForEncoding(value);
+        const result = maxDepth > 0 ? this.sanitizeForEncoding(value, maxDepth-1) : undefined;
```

Comment:
I wonder if stringifying the value would be better here:
```suggestion
        const result = maxDepth > 0 ? this.sanitizeForEncoding(value, maxDepth-1) : value.toString();
```

Reply:
>Date instances can be invalid & toISOString() will fail
ok but what about a date that can be invalid (do you have an example: I didn't find one in test) that will be embedded in the toString result?

Reply:
Oh I see, I didn't realize that invalid dates could pose a problem.

How about this in that case: let's move this maxDepth check above the `Object.entries(value)` iteration. There's no need to go over each key if we're planning to set them to `undefined` anyway. We can also directly return from there and issue the warning there as well. And we can return `value.toString()` which will just return `"[object Object"` which at least tell you that it's an object which we refused to serialize.

What do you think?

Reply:
I think this is a good way to fix this question, thanks 🙏